Classes/Service/NodeRedirectService.php

```diff
@@ -475,8 +482,14 @@ protected function getUriBuilder(): UriBuilder
             return $this->uriBuilder;
         }
 
-        $httpRequest = Request::createFromEnvironment();
-        $actionRequest = new ActionRequest($httpRequest);
+        /** @var HttpRequestHandlerInterface $requestHandler */
+        $requestHandler = $this->bootstrap->getActiveRequestHandler();
+        if (method_exists(ActionRequest::class, 'fromHttpRequest')) {
```

Comment:
Would be good to have a comment here, which line is needed for which version. Also to get it cleaned up at some point.